### DISCUSSION TITLE
what does i:ms:100 means

### DISCUSSION THREAD

**Discussion by wangjinnana** [bpftrace/bpftrace#1915] [Q&A]
I'm new to bpftrace, from https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md
there has an example, bpftrace supports C style while loops:
bpftrace -e 'i:ms:100 { $i = 0; while ($i <= 100) { printf("%d ", $i); $i++} exit(); }'

but i can't understand i:ms:100, could some one help me

**Comment by wangjinnana** [comment]
I found test cases in tests/parser.cpp, maybe it is shorter for interval:ms:100

```
TEST(Parser, while_loop)
{
  test("i:ms:100 { $a = 0; while($a < 10) { $a++ }}",
       R"PROG(Program
 interval:ms:100
  =
   variable: $a
   int: 0
  while(
   <
    variable: $a
    int: 10
   )
    variable: $a
     ++
)PROG");
}
```

**Comment by danobi** [comment]
Yes, `i` is the short name for `interval`. The full list of mappings can be found here: https://github.com/iovisor/bpftrace/blob/b2774221b687d543d672e7a9fb501b6d6382450d/src/types.h#L422-L440
